django_tasks/task.py

```diff
@@ -226,21 +226,37 @@ class TaskResult(Generic[T]):
     backend: str
     """The name of the backend the task will run on"""
 
-    _result: Optional[Union[T, dict]] = field(init=False, default=None)
+    _result: Optional[Union[T, SerializedExceptionDict]] = field(
+        init=False, default=None
+    )
 
     @property
     def result(self) -> Optional[Union[T, BaseException]]:
         if self.status == ResultStatus.COMPLETE:
             return cast(T, self._result)
         elif self.status == ResultStatus.FAILED:
             return (
-                exception_from_dict(cast(dict, self._result))
+                exception_from_dict(cast(SerializedExceptionDict, self._result))
                 if self._result is not None
                 else None
             )
 
         raise ValueError("Task has not finished yet")
 
+    @property
+    def traceback(self) -> Optional[str]:
+        """
+        Return the string representation of the traceback of the task if it failed
+        """
+
+        if self._result is None:
```

Comment:
Suggestion: Use a single case for the `None` return:

```python
if self._result is None or self.status != ResultStatus.FAILED:
```

Reply:
Done, but I inverted the logic, so that the specific working case return the value, otherwise, anything else returns None.